app/src/main/kotlin/com/imashnake/animite/features/home/HomeScreen.kt

```diff
@@ -144,15 +146,24 @@ fun HomeScreen(
 
                                 Row(
                                     modifier = bannerModifier.drawWithCache {
-                                        with(shader) {
-                                            setFloatUniform("resolution", size.width, size.height)
-                                            setFloatUniform("time", time.floatValue)
-                                            setColorUniform("orb", Color(0xFF6C408D).toArgb())
-                                            setColorUniform("bg", android.graphics.Color.TRANSPARENT)
-                                        }
-                                        onDrawBehind {
-                                            drawRect(ShaderBrush(shader))
-                                        }
+                                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
```

Comment:
Can't just `if shader != null` here?

Reply:
No, we have to return `DrawResult` in `drawWithCache`

Reply:
I mean we have an if/else here already, I was wondering if we can replace that with a check on `shader` instead